xdr/next/Stellar-contract.x

```diff
@@ -71,16 +71,115 @@ enum SCStatic
 enum SCStatusType
 {
     SST_OK = 0,
-    SST_UNKNOWN_ERROR = 1
+    SST_UNKNOWN_ERROR = 1,
+    SST_HOST_VALUE_ERROR = 2,
+    SST_HOST_OBJECT_ERROR = 3,
+    SST_HOST_FUNCTION_ERROR = 4,
+    SST_HOST_STORAGE_ERROR = 5,
+    SST_HOST_CONTEXT_ERROR = 6,
+    SST_VM_ERROR = 7
     // TODO: add more
 };
 
+enum SCHostValErrorCode
+{
+    HOST_VALUE_UNKNOWN_ERROR = 0,
+    RESERVED_TAG_VALUE = 1,
+    UNEXPECTED_VAL_TYPE = 2,
+    U63_OUT_OF_RANGE = 3,
+    U32_OUT_OF_RANGE = 4,
+    STATIC_UNKNOWN = 5,
+    MISSING_OBJECT = 6,
+    SYMBOL_TOO_LONG = 7,
+    SYMBOL_BAD_CHAR = 8,
+    SYMBOL_CONTAINS_NON_UTF8 = 9,
+    BITSET_TOO_MANY_BITS = 10,
+    STATUS_UNKNOWN = 11
+};
+
+enum SCHostObjErrorCode
+{
+    HOST_OBJECT_UNKNOWN_ERROR = 0,
+    UNKNOWN_HOST_OBJECT_REFERENCE = 1,
+    UNEXPECTED_HOST_OBJECT_TYPE = 2,
+    OBJECT_HANDLE_EXCEEDS_U32_MAX = 3,
```

Comment:
How can this error occur given that object handles are bound to u32 max?

Reply:
Internally when a new host object is added, it gets pushed into `Vec<HostObj>` stored in the `host`. The handle is the length of the Vec, which may exceed u32 max. This error check ensures it doesn't, then we cast the handle to `u32`. See [code ref](https://github.com/stellar/rs-stellar-contract-env/blob/main/stellar-contract-env-host/src/host.rs#L443-L445).

Reply:
Updated it to `HOST_OBJECT_OBJECT_COUNT_EXCEEDS_U32_MAX`.